backend/src/dao/TeamEventAttendanceDao.ts

```diff
@@ -1 +1,44 @@
-export default class TeamEventAttendanceDao {}
+import { v4 as uuidv4 } from 'uuid';
+import { memberCollection, teamEventAttendanceCollection } from '../firebase';
+import { DBTeamEventAttendance } from '../types/DataTypes';
+
+export default class TeamEventAttendanceDao {
+  static async createTeamEventAttendance(
+    teamEventAttendance: TeamEventAttendance
+  ): Promise<TeamEventAttendance> {
+    const teamAttendanceRef: DBTeamEventAttendance = {
+      member: memberCollection.doc(teamEventAttendance.member.email),
+      ...(typeof teamEventAttendance.hoursAttended === 'number'
+        ? { hoursAttended: teamEventAttendance.hoursAttended }
```

Comment:
I would not write the logic this way. `hoursAttended` is an optional field, so it can either be `number` or `undefined`. But that doesn't matter here, since you can just pass `teamEventAttendance.hoursAttended` directly to the database, whether it's undefined or it's `number`. I think by default that field, if you pass in `undefined` will just default to 0 in the DB, which is fine.

Reply:
i tried this originally, but when u pass in undefined throws an error saying cant use 'undefined' in firebase. But i can set it to zero if it is undefined

Reply:
Got it, makes sense. Personally I'd just pass in 0, or maybe `null` if that's allowed (we'd just have to change the types then since right now it's `number? = number | undefined`). And I'd use the nil-coalescing operator operator (`??`) for that to keep the logic short and simple to look at.

I want to hear from @JacksonStaniec @henry-li-06 though. Could you guys weigh in please?

Reply:
Oh, you know what? I see the issue. You can pass in `null`, not `undefined` into firebase. But I set the type to be `number | undefined`.

Reply:
The issue here is that Firebase currently does accept `undefined` values. But that's something we can change in a firebase config file. Once that's done, passing undefined to the DB will no longer lead to an error being thrown.

Reply:
got it, thanks for all the help!

Reply:
@kembayeb after [this PR](https://github.com/cornell-dti/idol/pull/426) gets merged in, rebase your branch on top of staging and then you can get rid of this for this method and for the update method

Reply:
Sure thing.

Jackson just got that fix in. Could you merge main into this branch and try it again, this time with `undefined` values?